### PR TITLE
Fix android build.

### PR DESCRIPTION
**Pull request by aaron112**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -38,7 +38,7 @@ else()
 	message(FATAL_ERROR "Unknown processor:" ${CMAKE_SYSTEM_PROCESSOR})
 endif()
 
-if (${CMAKE_GENERATOR} MATCHES Xcode AND (${ARCH} STREQUAL "arm" OR ${ARCH} STREQUAL "aarch64"))
+if (${BORINGSSL_NO_ASM})
 	# This is meant to capture the iOS case; the clang integrated assembler
```
Comment (thughes) [inline CMakeLists.txt]: Can you update the comment about iOS to include android?

File: tool/speed.cc [hunk 2]
```diff
@@ -169,8 +169,8 @@ struct free_functor {
   }
 };
 
-#if defined(OPENSSL_WINDOWS)
-#define AllocAligned malloc
+#if defined(OPENSSL_WINDOWS) || defined(OPENSSL_ANDROID)
```
Comment (thughes) [inline tool/speed.cc]: Looks like you commented out the `OPENSSL_ANDROID` definition in the cmake file. Is this still needed?

Comment (aaron112) [inline tool/speed.cc]: @bstrong Do you think this is needed? I commented out OPENSSL_ANDROID in cmake file and it's still working.

Comment (aaron112) [inline tool/speed.cc]: Tested without these and it's working. Removing.

File: CMakeLists.txt (hunk 1 above)
Comment (aaron112) [inline CMakeLists.txt]: Will do.

**Pull request merged**